CSharp/Library/Microsoft.Bot.Connector.Shared/SkillAuthentication/DefaultAllowedCallersClaimsValidator.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.SkillAuthentication
+{
+    /// <summary>
+    /// Claims validator which checks that requests are coming only from allowed parent bots.
+    /// </summary>
+    public class DefaultAllowedCallersClaimsValidator : ClaimsValidator
+    {
+        private readonly IList<string> _allowedCallers;
+
+        public DefaultAllowedCallersClaimsValidator(IList<string> allowedCallers)
+        {
+            // To allow all Parent bots, AllowedCallers should have one element of '*'
+
+            _allowedCallers = allowedCallers ?? throw new ArgumentNullException(nameof(allowedCallers));
+            if (_allowedCallers.Count == 0)
```

Comment:
I like `.Any()` more than count == 0, but it has zero runtime impact, just a personal nit that you can ignore if you want to :)

Reply:
Hmmm, i thought .Count on an IList had better performance: https://stackoverflow.com/a/305156/86646

Changed to .Any()